src/main/java/seedu/address/model/module/Prerequisite.java

```diff
@@ -0,0 +1,138 @@
+package seedu.address.model.module;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.LinkedList;
+
+/**
+ * Represents a Module's prerequisites in the system.
+ * */
+public class Prerequisite {
+    private LinkedList<LinkedList<Module>> prerequisites;
```

Comment:
This might not be robust enough for [certain module's](https://nusmods.com/courses/CS3282/thematic-systems-project-ii) prerequisite trees.